lib/components/mobile/batch-results-screen.js

```diff
@@ -0,0 +1,142 @@
+import Map from '../map/map'
+import Icon from '../narrative/icon'
+import NarrativeItineraries from '../narrative/narrative-itineraries'
+import {
+  getActiveItineraries,
+  getActiveSearch,
+  getResponsesWithErrors
+} from '../../util/state'
+
+import MobileContainer from './container'
+import ResultsError from './results-error'
+import ResultsHeader from './results-header'
+
+const StyledMobileContainer = styled(MobileContainer)`
+  .options > .header {
+    margin: 10px;
+  }
+
+  &.otp.mobile .mobile-narrative-container {
+    bottom: 0;
+    left: 0;
+    overflow-y: auto;
+    padding: 0;
+    position: fixed;
+    right: 0;
+  }
+`
+
+const ExpandMapButton = styled(Button)`
+  bottom: 10px;
+  left: 10px;
+  position: absolute;
+  z-index: 999999;
+`
+
+const VIEW = uiActions.ItineraryView
+
+const NARRATIVE_SPLIT_TOP_PERCENT = 45
+
+/**
+ * This component renders the mobile view of itinerary results from batch routing,
+ * and features a split view between the map and itinerary results or narratives.
+ */
+class BatchMobileResultsScreen extends Component {
+  componentDidUpdate (prevProps) {
+    if (this.props.activeLeg !== prevProps.activeLeg) {
+      // Check if the active leg has changed. If a different leg is selected,
+      // unexpand the itinerary to show the map focused on the selected leg
+      // (similar to the behavior of LineItinerary).
+      this._setItineraryExpanded(false)
+    }
+  }
+
+  _setItineraryExpanded = itineraryExpanded => {
+    this.props.setItineraryView(itineraryExpanded ? VIEW.FULL : VIEW.SPLIT)
+  }
+
+  _toggleMapExpanded = () => {
+    this.props.setItineraryView(this.props.itineraryView === VIEW.HIDDEN ? VIEW.SPLIT : VIEW.HIDDEN)
+  }
+
+  render () {
+    const { errors, itineraries, itineraryView } = this.props
+    const hasNoResult = itineraries.length === 0 && errors.length > 0
+    const mapExpanded = itineraryView === VIEW.HIDDEN
+    const itineraryExpanded = itineraryView === VIEW.FULL
+    const narrativeTop = mapExpanded ? '100%' : (itineraryExpanded ? '100px' : `${NARRATIVE_SPLIT_TOP_PERCENT}%`)
+    const mapBottom = mapExpanded ? 0 : `${100 - NARRATIVE_SPLIT_TOP_PERCENT}%`
+
+    return (
+      <StyledMobileContainer>
+        <ResultsHeader />
+        <div
+          className='results-map'
+          style={{bottom: mapBottom, display: itineraryExpanded ? 'none' : 'inherit'}}
+        >
+          <Map />
+          <ExpandMapButton
+            bsSize='small'
+            onClick={this._toggleMapExpanded}
+          >
+            <Icon name={mapExpanded ? 'list-ul' : 'arrows-alt'} />{' '}
+            {mapExpanded ? 'Show results' : 'Expand map'}
+          </ExpandMapButton>
+        </div>
+        {hasNoResult
+          ? <ResultsError
+            error={errors[0]}
+            style={{
+              display: mapExpanded ? 'none' : 'inherit',
+              top: narrativeTop,
+              transition: 'top 300ms'
+            }} />
```

Comment:
If ResultsError is changed to be styled-components-compatible (see https://github.com/opentripplanner/otp-react-redux/pull/339/files#r602670869), then this ResultsError component could be wrapped as a styled component that accepts props and styles accordingly. [See these docs](https://styled-components.com/docs/basics#adapting-based-on-props).

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/339/commits/d7737831bffd7e6a4f5294541b0187c1a2574c3a